test/unittests/xmlupload/test_project_client_live.py

```diff
@@ -0,0 +1,71 @@
+from dataclasses import dataclass, field
+from test.unittests.helpers.connection_mock import ConnectionMockBase
+from typing import Any
+
+from dsp_tools.utils.xmlupload.project_client import ProjectClientLive
+
+# pylint: disable=missing-class-docstring,missing-function-docstring,unused-argument,redefined-outer-name
+
+
+@dataclass
+class ConnectionMock(ConnectionMockBase):
+    get_responses: list[dict[str, Any]] = field(default_factory=list)
+
+    def get(self, route: str, headers: dict[str, str] | None = None) -> dict[str, Any]:
+        return self.get_responses.pop(0)
+
+
+class TestProjectClientLive:
+    """Test the ProjectClientLive class."""
+
+    def test_get_project_iri(self) -> None:
+        project_iri_response = {"project": {"id": "http://www.example.org/projects#a"}}
+        project_ontologies_response = {"@id": "http://www.example.org/ontologies/a.1/v2"}
+        con = ConnectionMock([project_iri_response, project_ontologies_response])
+        project_client = ProjectClientLive(con, "")
+        project_iri = project_client.get_project_iri()
+        expected = "http://www.example.org/projects#a"
+        assert project_iri == expected
+
+    def test_get_ontology_iris(self) -> None:
+        project_iri_response = {"project": {"id": "http://www.example.org/projects#a"}}
+        project_ontologies_response = {"@id": "http://www.example.org/ontologies/a.1/v2"}
+        con = ConnectionMock([project_iri_response, project_ontologies_response])
+        project_client = ProjectClientLive(con, "")
+        ontology_iris = project_client.get_ontology_iris()
+        expected = ["http://www.example.org/ontologies/a.1/v2"]
+        assert ontology_iris == expected
+
+    def test_get_ontology_name_dict(self) -> None:
+        project_iri_response = {"project": {"id": "http://www.example.org/projects#a"}}
+        project_ontologies_response = {
+            "@graph": [
+                {"@id": "http://www.example.org/ontologies/a.1/v2"},
+                {"@id": "http://www.example.org/ontologies/a.2/v2"},
+            ]
+        }
+        con = ConnectionMock([project_iri_response, project_ontologies_response])
+        project_client = ProjectClientLive(con, "")
+        ontology_name_dict = project_client.get_ontology_name_dict()
+        expected = {
+            "a.1": "http://www.example.org/ontologies/a.1/v2",
```

Comment:
Does that mean that "a.2" is automatically derived from "http://www.example.org/ontologies/a.1/v2"?

Reply:
yes